src/main/java/org/jvnet/solaris/libzfs/jna/libzfs.java

```diff
@@ -287,7 +288,10 @@ int zfs_prop_get_numeric(zfs_handle_t handle, zfs_prop_t prop, LongByReference r
 int zfs_iter_children(zfs_handle_t handle, zfs_iter_f callback, Pointer arg);
 int zfs_iter_dependents(zfs_handle_t handle, boolean _2, zfs_iter_f callback, Pointer arg);
 int zfs_iter_filesystems(zfs_handle_t handle, zfs_iter_f callback, Pointer arg);
+/* The legacy (Sun/Oracle Solaris; OpenSolaris) function ABI signature: */
```

Comment:
Question is probably linked to "the case of the unsupported value" above, so killing two birds with one message here... Once upon a time there was Sun, and its ZFS. Then there was OpenSolaris. Then there was the Lawnmower making open code proprietary again, and the illumos to fork off the latest legally open code. And OpenZFS chopping off the illumos OS-specific code and leave just the ZFS code. And the illumos-gate, ZoL and BSD and maybe MacOS projects becoming downstreams of OpenZFS each with its random schedule of adding innovations and porting changes from common OpenZFS upstream.

Given that the `libzfs.so.*` were OS-private libs, an implementation detail between SunOS kernel and systems management tools that lived in the same repo, API and ABI stability was not a concern for initial developers. Yet, this is about the only shared lib that allows dynamic linking to get into the run-time engine of ZFS on the host OS. As a consequence, there are many ABIs present in real-life distros that ALL must be handled without segfaulting the JVM due to bad signature ;) And noting the bug that currently ZFS support is hardcoded to be on "SunOS" platforms, when this check is fixed - there will be a lot more platforms and ABIs to handle (BSDs, Linuxii et al), with the nuance that each supported kernel and distro may have a different mix of function signatures currently supported at its run-time - ideally all or some of this might be guessed during startup, but for definite results (and/or fallback when guesswork is implemented but guesses wrong on a particular setup) we support envvar/prop configuration here.

So, it is neither really deprecated nor legacy - these are just two known signatures for this function that are present in the wild. Commits in this repo's history uncover many more such differences that should get handled, and it is likely (over time at least) that there will be more than two signatures for some functions.